ssh/tailssh/tailssh.go

```diff
@@ -1129,18 +1146,22 @@ func (c *conn) matchRule(r *tailcfg.SSHRule) (a *tailcfg.SSHAction, localUser st
 	if c.ruleExpired(r) {
 		return nil, "", nil, errRuleExpired
 	}
+	var userMatchError error
```

Comment:
This is a bit convoluted, but it allowed me to maintain the current behavior that we don't check the principal if the user failed to match, while still being able to identify when the principal is allowed but the specific user isn't.